File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/GoalHorizon.kt
package com.goalmaker.app.application.planning

/** The period a goal belongs to, longest first; [id] is how the server and the vectors name it. */
enum class GoalHorizon(val id: String, val rank: Int) {
    YEAR("year", 3),
    MONTH("month", 2),
    WEEK("week", 1),
    DAY("day", 0),
    ;

    companion object {
        fun of(id: String?): GoalHorizon? = entries.firstOrNull { it.id == id }
    }
}

File: windows/src/GoalMaker.Core/Planning/GoalHorizon.cs
namespace GoalMaker.Core.Planning;

/// <summary>The period a goal belongs to; the value is its rank, longest highest (docs/goals.md).</summary>
public enum GoalHorizon
{
    Day = 0,
    Week = 1,
    Month = 2,
    Year = 3,
}
